pkg/controller/docdb/dbinstance/setup.go

```diff
@@ -0,0 +1,220 @@
+	e.filterList = filterList
+}
+
+type hooks struct {
+	client docdbiface.DocDBAPI
+	kube   client.Client
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.DescribeDBInstancesInput) error {
+	obj.DBInstanceIdentifier = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBInstance, resp *svcsdk.DescribeDBInstancesOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclient.StringValue(cr.Status.AtProvider.DBInstanceStatus) {
+	case svcapitypes.DocDBInstanceStateAvailable:
+		cr.Status.SetConditions(xpv1.Available())
+	case svcapitypes.DocDBInstanceStateCreating:
+		cr.Status.SetConditions(xpv1.Creating())
+	case svcapitypes.DocDBInstanceStateDeleting:
+		cr.Status.SetConditions(xpv1.Deleting())
+	default:
+		cr.Status.SetConditions(xpv1.Unavailable())
+	}
+
+	obs.ConnectionDetails = getConnectionDetails(cr)
+	return obs, nil
+}
+
+func (e *hooks) isUpToDate(cr *svcapitypes.DBInstance, resp *svcsdk.DescribeDBInstancesOutput) (bool, error) { // nolint:gocyclo
+	instance := resp.DBInstances[0]
+
+	switch {
+	case awsclient.BoolValue(cr.Spec.ForProvider.AutoMinorVersionUpgrade) != awsclient.BoolValue(instance.AutoMinorVersionUpgrade),
+		awsclient.StringValue(cr.Spec.ForProvider.CACertificateIdentifier) != awsclient.StringValue(instance.CACertificateIdentifier),
+		awsclient.StringValue(cr.Spec.ForProvider.DBInstanceClass) != awsclient.StringValue(instance.DBInstanceClass),
+		awsclient.StringValue(cr.Spec.ForProvider.PreferredMaintenanceWindow) != awsclient.StringValue(instance.PreferredMaintenanceWindow),
+		awsclient.Int64Value(cr.Spec.ForProvider.PromotionTier) != awsclient.Int64Value(instance.PromotionTier):
+		return false, nil
+	}
+
+	return svcutils.AreTagsUpToDate(e.client, cr.Spec.ForProvider.Tags, instance.DBInstanceArn)
+}
+
+func lateInitialize(cr *svcapitypes.DBInstanceParameters, resp *svcsdk.DescribeDBInstancesOutput) error {
+	instance := resp.DBInstances[0]
+
+	cr.AvailabilityZone = awsclient.LateInitializeStringPtr(cr.AvailabilityZone, instance.AvailabilityZone)
+	cr.AutoMinorVersionUpgrade = awsclient.LateInitializeBoolPtr(cr.AutoMinorVersionUpgrade, instance.AutoMinorVersionUpgrade)
+	cr.CACertificateIdentifier = awsclient.LateInitializeStringPtr(cr.CACertificateIdentifier, instance.CACertificateIdentifier)
+	cr.PreferredMaintenanceWindow = awsclient.LateInitializeStringPtr(cr.PreferredMaintenanceWindow, instance.PreferredMaintenanceWindow)
+	cr.PromotionTier = awsclient.LateInitializeInt64Ptr(cr.PromotionTier, instance.PromotionTier)
+	return nil
+}
+
+func preUpdate(ctx context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.ModifyDBInstanceInput) error {
+	obj.CACertificateIdentifier = cr.Spec.ForProvider.CACertificateIdentifier
+	obj.ApplyImmediately = cr.Spec.ForProvider.ApplyImmediately
+	return nil
```

Comment:
I think you need to set `obj.DBInstanceIdentifier = awsclient.String(meta.GetExternalName(cr))` here, too.

Reply:
Fixed